### PR TITLE
My Site Dashboard - Phase 2 - Initial Structure -  Refresh layout

### PR DESCRIPTION
**Pull request by zwarm**
Parent #15215

This goal of this PR is to introduce the pull-to-refresh layout changes to the My Site view controlled by the `MySiteDashboardPhase2FeatureConfig` feature flag. This PR does not include any logic to actually refresh the sources.

**Notes**
- I split the @Before setUp in MySiteViewModelTest to support testing the PTR enabled flag. setUp now calls init() with a parameter for setting the MySiteDashboardPhase2FeatureConfig feature flag. If there is a better way, I am open to exploring it.
- The refresh indicator will remain showing. This feature will be implemented in a future PR. As it's behind a feature flag, this is okay for now. I can always add in temporary logic to dismiss it. lmk.

**Merge Instructions**
- Ensure that PR #15501 has been merged
- Remove Not Ready for Merge label
- Merge as normal

**To test:**
- Launch the app with MySiteDashboardPhase2FeatureConfig set to off
- Try to swipe to refresh
- Note: swipe to refresh is not active
- Navigate to My Site -> App Settings -> Debug Settings and set MySiteDashboardPhase2FeatureConfig to on
- Restart the App
- Navigate back to My Site tab
- Try to swipe to refresh
- Note: swipe to refresh indicator is shown and a snackbar notes that the action was requested

## Regression Notes
1. Potential unintended areas of impact
N/A -Logic is behind a feature flag

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing

3. What automated tests I added (or what prevented me from doing so)
- Added refresh tests to MySiteViewModelTest

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by ashiagr — APPROVED** [review]
Thanks, @zwarm! I've tested the changes, all looks good to me. 

PS: I noticed that the pull-to-refresh loading indicator doesn't go away. This is in accordance with our finding that there's no way to know the refresh completion state as of now and it will be tackled separately as we progress with our refactoring.

I'm approving the PR, feel free to merge once #15501 is merged.

**Comment by zwarm** [comment]
@ashiagr - thanks for the review.  And that's a yes on the refresh indicator - yet to be solved, but okay in this PR because it's behind the flag. 🙇

**Pull request merged**
